uaclient/entitlements/tests/test_fips.py

```diff
@@ -143,14 +142,15 @@ def test_enable_returns_false_on_missing_suites_directive(
     ):
         """When directives do not contain suites returns false."""
         # Unset suites directive
-        fips_entitled_no_suites = copy.deepcopy(
-            machine_access(entitlement.name)
-        )
-        fips_entitled_no_suites["entitlement"]["directives"]["suites"] = []
-        entitlement.cfg.write_cache(
-            "machine-access-{}".format(entitlement.name),
-            fips_entitled_no_suites,
+        token = entitlement.cfg.read_cache("machine-token")
+        entitlement_cfg = entitlement.cfg.entitlements[entitlement.name].get(
+            "entitlement"
         )
+        entitlement_cfg["directives"]["suites"] = []
+        token["machineTokenInfo"]["contractInfo"]["resourceEntitlements"] = [
+            entitlement_cfg
+        ]
+        entitlement.cfg.write_cache("machine-token", token)
```

Comment:
Similar to the other test we discussed, would this be simpler by using an entitlement factory directly? I think so:

```diff
diff --git a/uaclient/entitlements/tests/test_fips.py b/uaclient/entitlements/tests/test_fips.py
index a44672e..6f699d5 100644
--- a/uaclient/entitlements/tests/test_fips.py
+++ b/uaclient/entitlements/tests/test_fips.py
@@ -4,6 +4,7 @@ import contextlib
 import copy
 import itertools
 import mock
+from functools import partial
 
 import pytest
 
@@ -24,9 +25,14 @@ M_GETPLATFORM = M_REPOPATH + "util.get_platform_info"
 
 
 @pytest.fixture(params=[FIPSEntitlement, FIPSUpdatesEntitlement])
-def entitlement(request, entitlement_factory):
+def fips_entitlement_factory(request, entitlement_factory):
     """Parameterized fixture so we apply all tests to both FIPS and Updates"""
-    return entitlement_factory(request.param)
+    return partial(entitlement_factory, request.param)
+
+
+@pytest.fixture
+def entitlement(fips_entitlement_factory):
+    return fips_entitlement_factory()
 
 
 class TestFIPSEntitlementCanEnable:
@@ -138,19 +144,10 @@ class TestFIPSEntitlementEnable:
         "uaclient.util.get_platform_info", return_value={"series": "xenial"}
     )
     def test_enable_returns_false_on_missing_suites_directive(
-        self, m_platform_info, m_add_apt, entitlement
+        self, m_platform_info, m_add_apt, fips_entitlement_factory
     ):
         """When directives do not contain suites returns false."""
-        # Unset suites directive
-        token = entitlement.cfg.read_cache("machine-token")
-        entitlement_cfg = entitlement.cfg.entitlements[entitlement.name].get(
-            "entitlement"
-        )
-        entitlement_cfg["directives"]["suites"] = []
-        token["machineTokenInfo"]["contractInfo"]["resourceEntitlements"] = [
-            entitlement_cfg
-        ]
-        entitlement.cfg.write_cache("machine-token", token)
+        entitlement = fips_entitlement_factory(suites=[])
 
         with mock.patch.object(entitlement, "can_enable", return_value=True):
             with pytest.raises(exceptions.UserFacingError) as excinfo:
```

Reply:
(Oh, FYI, this patch won't apply cleanly without first applying the one above.)